lib/liquid/utils.rb

```diff
@@ -17,7 +17,7 @@ def self.slice_collection_using_each(collection, from, to)
       index = 0
 
       # Maintains Ruby 1.8.7 String#each behaviour on 1.9
-      return [collection] if non_blank_string?(collection)
+      return collection != ''.freeze ? [collection] : [] if collection.is_a?(String)
```

Comment:
I find this line hard to read (the combination of a `if` suffix and a ternary). Can you make this a bit more explicit?

Maybe
```ruby
if collection.is_a?(String)
  return collection != ''.freeze ? [collection] : []
end
```